src/solvers/strings/string_refinement.cpp

```diff
@@ -1410,7 +1396,7 @@ static std::pair<bool, std::vector<exprt>> check_axioms(
 
     stream << std::string(2, ' ') << i << ".\n";
     const exprt with_concretized_arrays =
-      substitute_array_access(negaxiom, gen_symbol, true);
+      substitute_array_access(negaxiom, generator.fresh_symbol, true);
```

Comment:
I was looking in our code base what this actually is and just to make sure I understand it right - we shouldn't make it `const` because it is an instance of `symbol_generatort` and by calling the `()` operator we actually change the underlying instance (by increasing the `symbol_count`). Right? Because we had it `const` it was behaving unexpectedly and probably not updating the count, thus resulting in multiple symbols of the same name.

Reply:
@svorenova exactly.